deployment/deploy-core-infrastructure.ps1

```diff
@@ -112,14 +131,109 @@ class Aks {
+      # Additional domains for observability
+      $dnsCustomDomainsHash = @{}
+      $dnsCustomDomainsHash.Add("gcr_$($dnsCustomDomainsHash.Count)", "gcr.io")
+      $dnsCustomDomainsHash.Add("jetstack_$($dnsCustomDomainsHash.Count)", "charts.jetstack.io")
+      $dnsCustomDomainsHash.Add("otel_$($dnsCustomDomainsHash.Count)", "open-telemetry.github.io")
+      $dnsCustomDomainsHash.Add("k8s_$($dnsCustomDomainsHash.Count)", "registry.k8s.io")
+      $dnsCustomDomainsHash.Add("k8s_$($dnsCustomDomainsHash.Count)", "cr.fluentbit.io")
+      $dnsCustomDomainsHelm = $dnsCustomDomainsHash.GetEnumerator() | ForEach-Object { "customDomains.$($_.Key)=$($_.Value)" }
+      $dnsCustomDomainsHelm = $dnsCustomDomainsHelm -Join ","
+
+      helm install dnsmasq azdistributededge/dnsmasqaks `
+        --set-string proxyDnsServer="$proxyIp" `
+        --set $dnsCustomDomainsHelm `
+        --namespace edge-infra `
+        --wait
+    }
+    else {
+      helm install dnsmasq azdistributededge/dnsmasqaks `
+        --set-string proxyDnsServer="$proxyIp" `
+        --namespace edge-infra `
+        --wait
+    }
 
     $dnsService = kubectl get service dsnmasq-service -n edge-infra -o json | ConvertFrom-Json
     $dnsMasqIp = $dnsService.status.loadBalancer.ingress.ip
 
+    # ----- Bootstrap monitoring and observability
+    if($installObservability) {
+
+      $metadataNetworkLayer = $aksName.Substring($aksName.Length - 2, 2)
+      Write-Title("Observability - install pre-req cert-manager for AKS $aksName")
+      helm repo add --force-update jetstack https://charts.jetstack.io
+      helm repo update
+
+      helm upgrade cert-manager jetstack/cert-manager `
+        --install `
+        --namespace cert-manager --create-namespace `
+        --set installCRDs=true `
+        --wait
+      
+      helm repo add open-telemetry https://open-telemetry.github.io/opentelemetry-helm-charts 
+      helm repo update
+
+      helm upgrade opentelemetry-operator open-telemetry/opentelemetry-operator `
+        --install `
+        --version "0.36.0" `
+        --namespace monitoring --create-namespace `
+        --wait
+
+      if($null -ne $parentProxyConfig) {
+        Write-Title("Install Helm chart with OpenTelemetry and FluentBit for AKS $aksName with parent collector")
+
+        $parentProxyIp = $parentProxyConfig.ProxyIp
+        $parentEndpoint = "http://${parentProxyIp}:4318"
+        helm upgrade otelcollection azdistributededge/otelcollection `
+        --install `
+        --namespace monitoring `
+        --set exporters.parentOtlp.enabled=true `
+        --set-string exporters.parentOtlp.endpoint="$parentEndpoint" `
+        --set-string metadata.region="$arcLocation" `
+        --set-string metadata.clusterName="$aksName" `
+        --set-string metadata.networkLayer="$metadataNetworkLayer" `
+        --wait 
+      }
+      else {
+        Write-Title("Install Helm chart with Observability with Grafana as visualization for AKS $aksName in top layer")
```

Comment:
maybe you wanted to write:
with Observability and Grafana???

Reply:
Done